File: server/src/com/broll/gainea/server/init/ExpansionSetting.java
package com.broll.gainea.server.init;

import com.broll.gainea.server.core.map.ExpansionType;

public enum ExpansionSetting {

    BASIC_GAME("Gainea", new ExpansionType[]{ExpansionType.GAINEA}),
    BOG("Eisländer", new ExpansionType[]{ExpansionType.ICELANDS}),
    ICE("Sumpf", new ExpansionType[]{ExpansionType.BOGLANDS}),
    PLUS_ICELANDS("Gainea & Eisländer", new ExpansionType[]{ExpansionType.GAINEA, ExpansionType.ICELANDS}),
    PLUS_ICELANDS_AND_BOG("Gainea & Eisländer & Sumpf", new ExpansionType[]{ExpansionType.GAINEA, ExpansionType.ICELANDS, ExpansionType.BOGLANDS}),
    PLUS_BOG("Gainea & Sumpf", new ExpansionType[]{ExpansionType.GAINEA, ExpansionType.BOGLANDS}),
    PLUS_ICELANDS_AND_MOUNTAINS("Gainea & Eisländer & Gipfel", new ExpansionType[]{ExpansionType.GAINEA, ExpansionType.ICELANDS, ExpansionType.MOUNTAINS}),
    PLUS_BOG_AND_MOUNTAINS("Gainea & Sumpf & Gipfel", new ExpansionType[]{ExpansionType.GAINEA, ExpansionType.BOGLANDS, ExpansionType.MOUNTAINS}),
    BOG_AND_ICE("Eisländer & Sumpf", new ExpansionType[]{ExpansionType.ICELANDS, ExpansionType.BOGLANDS}),
    FULL("Alle", ExpansionType.values());

    private String name;
    private ExpansionType[] maps;

    ExpansionSetting(String name, ExpansionType[] maps) {
        this.name = name;
        this.maps = maps;
    }

    public String getName() {
        return name+ " ("+getMaps().length+" - "+(getMaps().length * 2 +1)+" Spieler)";
    }

    public ExpansionType[] getMaps() {
        return maps;
    }
}

File: server/src/main/com/broll/gainea/server/init/ExpansionSetting.kt
package com.broll.gainea.server.init

import com.broll.gainea.server.core.map.ExpansionType

enum class ExpansionSetting(private override val name: String, val maps: Array<ExpansionType>) {
    BASIC_GAME("Gainea", arrayOf(ExpansionType.GAINEA)),
    BOG("Eisländer", arrayOf(ExpansionType.ICELANDS)),
    ICE("Sumpf", arrayOf(ExpansionType.BOGLANDS)),
    PLUS_ICELANDS("Gainea & Eisländer", arrayOf(ExpansionType.GAINEA, ExpansionType.ICELANDS)),
    PLUS_ICELANDS_AND_BOG("Gainea & Eisländer & Sumpf", arrayOf(ExpansionType.GAINEA, ExpansionType.ICELANDS, ExpansionType.BOGLANDS)),
    PLUS_BOG("Gainea & Sumpf", arrayOf(ExpansionType.GAINEA, ExpansionType.BOGLANDS)),
    PLUS_ICELANDS_AND_MOUNTAINS("Gainea & Eisländer & Gipfel", arrayOf(ExpansionType.GAINEA, ExpansionType.ICELANDS, ExpansionType.MOUNTAINS)),
    PLUS_BOG_AND_MOUNTAINS("Gainea & Sumpf & Gipfel", arrayOf(ExpansionType.GAINEA, ExpansionType.BOGLANDS, ExpansionType.MOUNTAINS)),
    BOG_AND_ICE("Eisländer & Sumpf", arrayOf(ExpansionType.ICELANDS, ExpansionType.BOGLANDS)),
    FULL("Alle", ExpansionType.entries.toTypedArray());

    fun getName(): String {
        return name + " (" + maps.size + " - " + (maps.size * 2 + 1) + " Spieler)"
    }
}
